src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -211,15 +211,19 @@ protected internal void RuntimeLoadScript(byte[] script, CallFlags callFlags, Ar
             if ((callFlags & ~CallFlags.All) != 0)
                 throw new ArgumentOutOfRangeException(nameof(callFlags));
 
-            ExecutionContextState state = CurrentContext.GetState<ExecutionContextState>();
-            ExecutionContext context = LoadScript(new Script(script, true), configureState: p =>
+            var state = CurrentContext.GetState<ExecutionContextState>();
+            if (state.IsDynamicCall && IsHardforkEnabled(Hardfork.HF_Echidna))
```

Comment:
It looks like this fix won't solve the problem completely. Consider the following situation: I deploy a contract that has a `loadDynamic` method that uses `System.Runtime.LoadScript` to load some script dynamically. Inside this dynamic script there's a call to the same `loadDynamic` method of the contract. So the invocation stack will look like `Contract` -> `Dynamic Script` -> `Contract` -> `DynamicScript` -> etc...

So `state.IsDynamicCall` will be `false` in case of the Contract's execution context.

Reply:
Yes, is not completely solved, but at least is an extra protection, also I would like to increse the price of this syscall

Reply:
And probably cache the script by hash, some time will spend parsing the script every time

Reply:
> at least is an extra protection

I don't think so. The root of the problem is fee imbalance. Just think of ExecutionFeeFactor=30, no problems there.

> And probably cache the script by hash, some time will spend parsing the script every time

I will just add a NOP to the script each time.

Reply:
If the script can fit into 32-4 bytes (likely it can, arguments are to be passed from the contract) I'd do `RET 0x00 0x00 0x00` and then `INC` it on every call, `INC` is cheap.

Reply:
Or CAT/SETITEM, I've just thought about LE and script correctness checks, the trick above may not work. Less optimal, I agree. Still, execution fee factor is the primary problem. We can raise this interop price, but https://github.com/neo-project/neo/issues/3552#issuecomment-2436165391